gramhopper/configuration/trigger_response_parser.py

```diff
@@ -0,0 +1,121 @@
+import abc
+from typing import Dict, get_type_hints, Union, List, Any, Type
+from inspect import isclass
+from .partial_ruamel_yaml import CommentedMap, CommentedSeq
+from .common_types import TriggerResponse, GlobalsDict
+from .boolean_helper import BooleanHelper
+from ..dict_enum import DictEnum
+
+
+class TriggerResponseParser(abc.ABC):
+    """ A common base parser for trigger/response configuration parsers"""
+
+    def __init__(self, mapping_class: Type[DictEnum], element_base_class: Type[TriggerResponse]):
+        self.__mapping_class = mapping_class
+        self.__element_base_class = element_base_class
+
+    @property
+    def mapping_class(self) -> Type[DictEnum]:
```

Comment:
That's the pythonic way to make it read-only?

Reply:
kind-of.
It's just a property with a getter and without a setter.